### ISSUE TITLE
Prometheus Receiver with TargetAllocator watching ServiceMonitors only adds Scrape Jobs but not actually scraping

### ISSUE THREAD

**Issue by changexd** [open-telemetry/opentelemetry-operator#2262]
### Component(s)

collector, target allocator

## Description
Following the tutorial for setting up TA with Prometheus serviceMonitor in [target allocator](https://github.com/open-telemetry/opentelemetry-operator/blob/main/cmd/otel-allocator/README.md)

1. the collector displays "Scrape job Added" along with serviceMonitor name, but it doesn't scrape anything except for its own static scrape config
2. TargetAllocator endpoint "/scrape_configs" outputs the converted serviceMonitor, but not displaying jobs in "/jobs"

## Steps to Reproduce
1. kubectl apply -f https://github.com/open-telemetry/opentelemetry-operator/releases/latest/download/opentelemetry-operator.yaml
2. apply resource from tutotrial with custom service-monitor
```
apiVersion: opentelemetry.io/v1alpha1
kind: OpenTelemetryCollector
metadata:
  name: prometheus-collector
  namespace: monitoring
spec:
  mode: statefulset
  targetAllocator:
    enabled: true
    prometheusCR:
      scrapeInterval: 5s
      enabled: true
  config: |
    receivers:
      prometheus:
        config:
          scrape_configs:
          - job_name: 'otel-collector'
            scrape_interval: 10s
            static_configs:
            - targets: [ '0.0.0.0:8888' ]
        target_allocator:
          endpoint: http://prometheus-collector-targetallocator
          interval: 5s
          collector_id: "${POD_NAME}"
    exporters:
      debug:

    service:
      pipelines:
        metrics:
          receivers: [prometheus]
          processors: []
          exporters: [debug]
```

## Expected Result
1. collector should be scraping jobs allocated by the targetAllocator and output to the console
2. target allocator endpoints "/jobs" should be displaying jobs from the "/scrape_configs" endpoint 

## Actual Result
1. collector only collects static target but did add scrape jobs
```
2023-10-24T06:54:23.610Z    info    service@v0.87.0/telemetry.go:84    Setting up own telemetry...
2023-10-24T06:54:23.610Z    info    service@v0.87.0/telemetry.go:201    Serving Prometheus metrics    {"address": ":8888", "level": "Basic"}
2023-10-24T06:54:23.610Z    info    exporter@v0.87.0/exporter.go:275    Development component. May change in the future.    {"kind": "exporter", "data_type": "metrics", "name": "debug"}
2023-10-24T06:54:23.611Z    info    service@v0.87.0/service.go:143    Starting otelcol...    {"Version": "0.87.0", "NumCPU": 4}
2023-10-24T06:54:23.611Z    info    extensions/extensions.go:33    Starting extensions...
2023-10-24T06:54:23.612Z    info    prometheusreceiver@v0.87.0/metrics_receiver.go:239    Starting discovery manager    {"kind": "receiver", "name": "prometheus", "data_type": "metrics"}
2023-10-24T06:54:23.612Z    info    prometheusreceiver@v0.87.0/metrics_receiver.go:105    Starting target allocator discovery    {"kind": "receiver", "name": "prometheus", "data_type": "metrics"}
2023-10-24T06:54:23.628Z    info    prometheusreceiver@v0.87.0/metrics_receiver.go:230    Scrape job added    {"kind": "receiver", "name": "prometheus", "data_type": "metrics", "jobName": "serviceMonitor/monitoring/kube-prometheus-stack-
kubelet/0"}
2023-10-24T06:54:23.628Z    info    prometheusreceiver@v0.87.0/metrics_receiver.go:230    Scrape job added    {"kind": "receiver", "name": "prometheus", "data_type": "metrics", "jobName": "serviceMonitor/monitoring/kube-prometheus-stack-
kubelet/1"}
2023-10-24T06:54:23.628Z    info    prometheusreceiver@v0.87.0/metrics_receiver.go:230    Scrape job added    {"kind": "receiver", "name": "prometheus", "data_type": "metrics", "jobName": "serviceMonitor/monitoring/kube-prometheus-stack-
prometheus-node-exporter/0"}
2023-10-24T06:54:23.628Z    info    prometheusreceiver@v0.87.0/metrics_receiver.go:230    Scrape job added    {"kind": "receiver", "name": "prometheus", "data_type": "metrics", "jobName": "serviceMonitor/monitoring/kube-prometheus-stack-
prometheus/0"}
2023-10-24T06:54:23.628Z    info    prometheusreceiver@v0.87.0/metrics_receiver.go:230    Scrape job added    {"kind": "receiver", "name": "prometheus", "data_type": "metrics", "jobName": "otel-collector"}
2023-10-24T06:54:23.628Z    info    prometheusreceiver@v0.87.0/metrics_receiver.go:230    Scrape job added    {"kind": "receiver", "name": "prometheus", "data_type": "metrics", "jobName": "serviceMonitor/monitoring/kube-prometheus-stack-
kubelet/2"}
2023-10-24T06:54:23.628Z    info    prometheusreceiver@v0.87.0/metrics_receiver.go:230    Scrape job added    {"kind": "receiver", "name": "prometheus", "data_type": "metrics", "jobName": "serviceMonitor/monitoring/kube-prometheus-stack-
operator/0"}
2023-10-24T06:54:23.628Z    info    service@v0.87.0/service.go:169    Everything is ready. Begin running and processing data.
2023-10-24T06:54:23.629Z    info    prometheusreceiver@v0.87.0/metrics_receiver.go:281    Starting scrape manager    {"kind": "receiver", "name": "prometheus", "data_type": "metrics"}
```
2. TA "/jobs" endpoint not displaying jobs from "/scrape_configs" as what the tutorial presents, instead displays only the static target
```
# result from /jobs
{"otel-collector":{"_link":"/jobs/otel-collector/targets"}}

# one example from /scrape_configs
"serviceMonitor/monitoring/kube-prometheus-stack-kubelet/0": {
    "authorization": {
      "credentials_file": "/var/run/secrets/kubernetes.io/serviceaccount/token",
      "type": "Bearer"
    },
    "enable_http2": true,
    "follow_redirects": true,
    "honor_labels": true,
    "honor_timestamps": true,
    "job_name": "serviceMonitor/monitoring/kube-prometheus-stack-kubelet/0",
    "kubernetes_sd_configs": [
      {
        "enable_http2": true,
        "follow_redirects": true,
        "kubeconfig_file": "/.kube/config",
        "namespaces": {
          "names": [
            "kube-system"
          ],
          "own_namespace": false
        },
        "role": "endpointslice"
      }
    ],
    "metrics_path": "/metrics",
    "relabel_configs": [
      {
        "action": "replace",
        "regex": "(.*)",
        "replacement": "$1",
        "separator": ";",
        "source_labels": [
          "job"
        ],
        "target_label": "__tmp_prometheus_job_name"
      },
      {
        "action": "keep",
        "regex": "(kubelet);true",
        "replacement": "$1",
        "separator": ";",
        "source_labels": [
          "__meta_kubernetes_service_label_app_kubernetes_io_name",
          "__meta_kubernetes_service_labelpresent_app_kubernetes_io_name"
        ]
      },
      {
        "action": "keep",
        "regex": "(kubelet);true",
        "replacement": "$1",
        "separator": ";",
        "source_labels": [
          "__meta_kubernetes_service_label_k8s_app",
          "__meta_kubernetes_service_labelpresent_k8s_app"
        ]
      },
      {
        "action": "keep",
        "regex": "https-metrics",
        "replacement": "$1",
        "separator": ";",
        "source_labels": [
          "__meta_kubernetes_endpointslice_port_name"
        ]
      },
      {
        "action": "replace",
        "regex": "Node;(.*)",
        "replacement": "${1}",
        "separator": ";",
        "source_labels": [
          "__meta_kubernetes_endpointslice_address_target_kind",
          "__meta_kubernetes_endpointslice_address_target_name"
        ],
        "target_label": "node"
      },
      {
        "action": "replace",
        "regex": "Pod;(.*)",
        "replacement": "${1}",
        "separator": ";",
        "source_labels": [
          "__meta_kubernetes_endpointslice_address_target_kind",
          "__meta_kubernetes_endpointslice_address_target_name"
        ],
        "target_label": "pod"
      },
      {
        "action": "replace",
        "regex": "(.*)",
        "replacement": "$1",
        "separator": ";",
        "source_labels": [
          "__meta_kubernetes_namespace"
        ],
        "target_label": "namespace"
      },
      {
        "action": "replace",
        "regex": "(.*)",
        "replacement": "$1",
        "separator": ";",
        "source_labels": [
          "__meta_kubernetes_service_name"
        ],
        "target_label": "service"
      },
      {
        "action": "replace",
        "regex": "(.*)",
        "replacement": "$1",
        "separator": ";",
        "source_labels": [
          "__meta_kubernetes_pod_name"
        ],
        "target_label": "pod"
      },
      {
        "action": "replace",
        "regex": "(.*)",
        "replacement": "$1",
        "separator": ";",
        "source_labels": [
          "__meta_kubernetes_pod_container_name"
        ],
        "target_label": "container"
      },
      {
        "action": "drop",
        "regex": "(Failed|Succeeded)",
        "replacement": "$1",
        "separator": ";",
        "source_labels": [
          "__meta_kubernetes_pod_phase"
        ]
      },
      {
        "action": "replace",
        "regex": "(.*)",
        "replacement": "${1}",
        "separator": ";",
        "source_labels": [
          "__meta_kubernetes_service_name"
        ],
        "target_label": "job"
      },
      {
        "action": "replace",
        "regex": "(.+)",
        "replacement": "${1}",
        "separator": ";",
        "source_labels": [
          "__meta_kubernetes_service_label_k8s_app"
        ],
        "target_label": "job"
      },
      {
        "action": "replace",
        "regex": "(.*)",
        "replacement": "https-metrics",
        "separator": ";",
        "target_label": "endpoint"
      },
      {
        "action": "replace",
        "regex": "(.*)",
        "replacement": "$1",
        "separator": ";",
        "source_labels": [
          "__metrics_path__"
        ],
        "target_label": "metrics_path"
      },
      {
        "action": "hashmod",
        "modulus": 1,
        "regex": "(.*)",
        "replacement": "$1",
        "separator": ";",
        "source_labels": [
          "__address__"
        ],
        "target_label": "__tmp_hash"
      },
      {
        "action": "keep",
        "regex": "$(SHARD)",
        "replacement": "$1",
        "separator": ";",
        "source_labels": [
          "__tmp_hash"
        ]
      }
    ],
    "scheme": "https",
    "scrape_interval": "5s",
    "scrape_timeout": "5s",
    "tls_config": {
      "ca_file": "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
      "insecure_skip_verify": true
    }
  },
}
```

### Kubernetes Version

1.24.10

### Operator version

0.87.0

### Collector version

0.87.0

**Comment by olehrgf** [comment]
I faced the same issue.
This issue starts from v0.86.0.
The cause of this is the default config for the kube file in the target allocator [cmd/otel-allocator/config/flags.go](https://github.com/swiatekm-sumo/opentelemetry-operator/blob/e7c65941522143c2c66ec12a49066bd1800084ee/cmd/otel-allocator/config/flags.go#L43C8-L43C8)
It must be empty by default.

As a workaround, you can specify an older image for the target allocator:

```
apiVersion: opentelemetry.io/v1alpha1
kind: OpenTelemetryCollector
metadata:
  name: prometheus-collector
spec:
  mode: statefulset
  replicas: 2
  targetAllocator:
    enabled: true
    image: ghcr.io/open-telemetry/opentelemetry-operator/target-allocator:0.85.0
    prometheusCR:
      enabled: true

```

**Comment by changexd** [comment]
@olehrgf  thank you so much for the information and workaround, it works now! After hours of digging, I did find the issue of kubeconfig file inside scrape configs, but thought it would simply use the kubernetes credential file inside the collector for kubernetes service discoveries. 

this is my very first issue, should I keep this issue opened until fixed?

If this is identified as a bug and needs a helping hand, it would be my honor to try fixing this(if this is also considered a good first issue).

**Comment by jaronoff97** [comment]
@changexd thanks for reporting this, we should definitely fix this before our next release.

**Comment by swiatekm-sumo** [comment]
Looks like this was introduced in https://github.com/open-telemetry/opentelemetry-operator/pull/1957. Indeed, I've missed one line in the Kube config parsing logic, we should set:

```go
target.KubeConfigFilePath = "" // reset as we use in cluster configuration
```

here: https://github.com/open-telemetry/opentelemetry-operator/blob/c7a96ac6b177b7e267147fb51d5ba6376e5a317c/cmd/otel-allocator/config/config.go#L98.

What's more worrying is that none of our automated tests caught this. I think it's time to actually add integration tests checking if the target allocator and Prometheus CR setup actually collects metrics. If @changexd would like to fix this issue, I'll add the integration tests, and hopefully we can see how their change makes the tests pass.

**Comment by jaronoff97** [comment]
Thanks for taking a look. @changexd let me know if you want to be assigned, if not we can find someone else :)

**Comment by changexd** [comment]
@jaronoff97 @swiatekm-sumo  I would like to be assigned for this issue, thank you!

**Comment by jaronoff97** [comment]
closed by #2275 and #2273. This will be in the next release. Thank you!
